zuul-core/src/test/java/com/netflix/zuul/netty/server/ServerTest.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ *      Licensed under the Apache License, Version 2.0 (the "License");
+ *      you may not use this file except in compliance with the License.
+ *      You may obtain a copy of the License at
+ *
+ *          http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *      Unless required by applicable law or agreed to in writing, software
+ *      distributed under the License is distributed on an "AS IS" BASIS,
+ *      WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *      See the License for the specific language governing permissions and
+ *      limitations under the License.
+ */
+
+package com.netflix.zuul.netty.server;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.mock;
+import com.netflix.netty.common.metrics.EventLoopGroupMetrics;
+import com.netflix.netty.common.status.ServerStatusManager;
+import com.netflix.spectator.api.Spectator;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.group.DefaultChannelGroup;
+import io.netty.util.concurrent.GlobalEventExecutor;
+import java.net.InetSocketAddress;
+import java.net.SocketAddress;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for {@link Server}.
+ */
+@RunWith(JUnit4.class)
+public class ServerTest {
+
+    @Test
+    public void getListeningSockets() {
+        ServerStatusManager ssm = mock(ServerStatusManager.class);
+        Map<SocketAddress, ChannelInitializer<?>> initializers = new HashMap<>();
+        initializers.put(
```

Comment:
Might want to init with at least 2 addrs. Weird things manifest themselves in collections of a single size 😄

Reply:
Good point, done.